x-pack/plugins/observability/public/pages/overview/data_sections.tsx

```diff
@@ -4,74 +4,46 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
 import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React from 'react';
+import { APMSection } from '../../components/app/section/apm';
 import { LogsSection } from '../../components/app/section/logs';
 import { MetricsSection } from '../../components/app/section/metrics';
-import { APMSection } from '../../components/app/section/apm';
 import { UptimeSection } from '../../components/app/section/uptime';
 import { UXSection } from '../../components/app/section/ux';
-import {
-  HasDataResponse,
-  ObservabilityFetchDataPlugins,
-  UXHasDataResponse,
-} from '../../typings/fetch_overview_data';
+import { ObservabilityHasDataResponse } from '../../typings/fetch_overview_data';
 
 interface Props {
-  bucketSize: string;
-  absoluteTime: { start?: number; end?: number };
-  relativeTime: { start: string; end: string };
-  hasData: Record<ObservabilityFetchDataPlugins, HasDataResponse>;
+  hasData?: Partial<ObservabilityHasDataResponse>;
 }
 
-export function DataSections({ bucketSize, hasData, absoluteTime, relativeTime }: Props) {
+export function DataSections({ hasData }: Props) {
   return (
     <EuiFlexItem grow={false}>
       <EuiFlexGroup direction="column">
         {hasData?.infra_logs && (
           <EuiFlexItem grow={false}>
-            <LogsSection
-              bucketSize={bucketSize}
-              absoluteTime={absoluteTime}
-              relativeTime={relativeTime}
-            />
+            <LogsSection />
           </EuiFlexItem>
         )}
         {hasData?.infra_metrics && (
           <EuiFlexItem grow={false}>
-            <MetricsSection
-              bucketSize={bucketSize}
-              absoluteTime={absoluteTime}
-              relativeTime={relativeTime}
-            />
+            <MetricsSection />
           </EuiFlexItem>
         )}
         {hasData?.apm && (
           <EuiFlexItem grow={false}>
-            <APMSection
-              bucketSize={bucketSize}
-              absoluteTime={absoluteTime}
-              relativeTime={relativeTime}
-            />
+            <APMSection />
           </EuiFlexItem>
         )}
         {hasData?.uptime && (
           <EuiFlexItem grow={false}>
-            <UptimeSection
-              bucketSize={bucketSize}
-              absoluteTime={absoluteTime}
-              relativeTime={relativeTime}
-            />
+            <UptimeSection />
           </EuiFlexItem>
         )}
         {hasData?.ux && (
           <EuiFlexItem grow={false}>
-            <UXSection
-              serviceName={(hasData.ux as UXHasDataResponse).serviceName as string}
-              bucketSize={bucketSize}
-              absoluteTime={absoluteTime}
-              relativeTime={relativeTime}
-            />
+            <UXSection serviceName={hasData.ux.serviceName as string} />
```

Comment:
qq: why does hasData for `ux` return data in a different format?

Reply:
@shahzad31 👆🏻